### ISSUE TITLE
Disable syntax highlighting by default, or on stuff that doesn't look like code

### ISSUE THREAD

**Issue by ara4n** [vector-im/vector-web#508]
A crappy heuristic like "needs a curly or right-angled bracket" might be sufficient for now.

**Comment by Xe** [comment]
Here is valid code that doesn't have curly or angle brackets:

```lua
print "hello world"

function hi(x)
    print("hi " .. x)
end 
```

```haskell
main :: IO ()
main = putStrLn "Hello world!"
```

**Comment by ara4n** [comment]
Suggestinons for better heuristics more than welcome - I just don't want 
to lose the autohighlighting entirely, as it's been useful more times than 
otherwise so far.  Perhaps highlight.js itself can be tuned.

**Comment by Xe** [comment]
does it support things like github flavored markdown language tagging?

    ```haskell
    main :: IO ()
    main = putStrLn "Hello World!"
    ```

```haskell
main :: IO ()
main = putStrLn "Hello World!"
```

**Comment by aperezdc** [comment]
GFM-style languge tagging for preformatted blocks can be done with something like (I have this [working for my webpage](https://github.com/aperezdc/perezdecastro.org/blob/master/site.js#L21-L25)):

```javascript
marked.setOptions({
	highlight: function (code, lang) {
		return (lang === undefined) ? code : H.highlight(lang, code).value;
	},
});
```

Note that the above will _disable highlighting for untagged preformatted code blocks_, so it might be desirable to apply some heuristic anyway when `lang === undefined`.

Should I send a PR for `matrix-react-sdk` adding this?

**Comment by ara4n** [comment]
sure  - PR would be very welcome.  The highlighting is currently done by highlight.js as a non-reactish DOM pass once the page has rendered.  It's completely decoupled from marked currently.  If there's a nice way to do make marked do the highlighting (whilst retaining highlight.js's heuristics if no language is marked) that'd be awesome.

https://github.com/matrix-org/matrix-react-sdk/blob/master/src/components/views/messages/TextualBody.js#L72 is the current super-stupid code.

**Comment by ara4n** [comment]
just realised that making marked do this would shift responsibility to the client sending the message rather than displaying it, which is a bit of a pain in the ass.  To fix this properly we'd need to finally get the "human representation of message" stuff landed in the spec to allow native markdown to be included in events, so the receiving client can then render the markdown as it liked.

I suspect a better bet would be to instead make the existing highlight.js code 'do the right thing' for syntax highlighting.

**Comment by aperezdc** [comment]
@ara4n: Yeah, I noticed also the shift in responsibility when tying this to `marked`. One idea would be to make the `<code>` elements have a `lang="javascript"` attribute added by `marked` when it renders the HTML, and then check for that and pass it to `highlight.js` when doing the syntax highlighting. WDYT?
